src/main/scala/org/ergoplatform/nodeView/state/UtxoStateReader.scala

```diff
@@ -38,22 +39,23 @@ trait UtxoStateReader extends ErgoStateReader with TransactionValidation {
     */
   def validateWithCost(tx: ErgoTransaction,
                        stateContextOpt: Option[ErgoStateContext],
-                       complexityLimit: Int,
+                       costLimit: Long,
                        interpreterOpt: Option[ErgoInterpreter]): Try[Long] = {
     val context = stateContextOpt.getOrElse(stateContext)
 
     val verifier = interpreterOpt.getOrElse(ErgoInterpreter(context.currentParameters))
 
     tx.statelessValidity().flatMap { _ =>
       val boxesToSpend = tx.inputs.flatMap(i => boxById(i.boxId))
-      val txComplexity = boxesToSpend.map(_.ergoTree.complexity).sum
-      if (txComplexity > complexityLimit) {
-        throw new Exception(s"Transaction $tx has too high complexity $txComplexity")
-      }
       tx.statefulValidity(
-        boxesToSpend,
```

Comment:
validateStateful internally uses MaxBlockCost as the limit. Now, when we have costLimit, it should be passed in validateStateful and take place of MaxBlockCost limit.

Reply:
see my reply below